hash_map.h

```diff
@@ -0,0 +1,171 @@
+//
+// Created by semyon on 03.02.2020.
+//
+
+#ifndef LONG_CONTEST_HASHMAP_H
+#define LONG_CONTEST_HASHMAP_H
+
+#include <memory>
+#include <vector>
+#include <list>
+
+template<class KeyType, class ValueType, class Hash = std::hash<KeyType> >
+class HashMap {
+public:
+    using iterator = typename std::list<std::pair<const KeyType, ValueType>>::iterator;
+    using const_iterator = typename std::list<std::pair<const KeyType, ValueType>>::const_iterator;
+    using pair_type = std::pair<const KeyType, ValueType>;
+    using pair_type_nonconst = std::pair<KeyType, ValueType>;
+    const std::size_t size_default = 1024;
+
+    iterator begin() noexcept {
+        return nodes.begin();
+    }
+
+    const_iterator begin() const noexcept {
+        return nodes.begin();
+    }
+
+    iterator end() noexcept {
+        return nodes.end();
+    }
+
+    const_iterator end() const noexcept {
+        return nodes.end();
+    }
+
+    HashMap(Hash hasher = Hash()) : size_table(size_default), hasher(hasher), table(size_default) {}
+
+    HashMap(std::initializer_list<pair_type_nonconst> init, Hash hasher = Hash()) : HashMap() {
```

Comment:
please use constructor delegation, not custom function

Reply:
fixed